torch_geometric/nn/encoding.py

```diff
@@ -94,3 +97,193 @@ def forward(self, x: Tensor) -> Tensor:
+        out_channels (int): Size of each output sample.
+        is_sorted (bool, optional): If set to :obj:`True`, assumes that
+            :obj:`edge_index` is sorted by column. This avoids internal
+            re-sorting of the data and can improve runtime and memory
+            efficiency. (default: :obj:`False`)
+        dropout (float, optional): Dropout probability of the MLP layer.
+            (default: :obj:`0.5`)
+
+    """
+    def __init__(
+        self,
+        K: int,
+        in_channels: int,
+        hidden_channels: int,
+        out_channels: int,
+        time_channels: int,
+        is_sorted: bool = False,
+        dropout: float = 0.5,
+    ) -> None:
+        super().__init__()
+        self.K = K
+        self.in_channels = in_channels
+        self.hidden_channels = hidden_channels
+        self.out_channels = out_channels
+        self.time_channels = time_channels
+        self.is_sorted = is_sorted
+        self.dropout = dropout
+
+        # teomporal encoder
+        self.temporal_encoder = TemporalEncoding(time_channels)
+        self.temporal_encoder_head = torch.nn.Linear(
+            time_channels + in_channels,
+            hidden_channels,
+        )
+
+        # temporal information summariser
+        self.mlp_mixer = _MLPMixer(
+            num_tokens=K,
+            in_channels=hidden_channels,
+            out_channels=out_channels,
+            dropout=dropout,
+        )
+
+    def forward(
+        self,
+        edge_attr: Tensor,
+        edge_time: Tensor,
+        edge_index: Tensor,
+        num_nodes: Optional[int] = None,
+    ) -> Tensor:
+        """
+        Args:
+            edge_attr (torch.Tensor): The edge features of shape
+                :obj:`[num_edges, in_channels]`.
+            edge_time (torch.Tensor): The time tensor of shape
+                :obj:`[num_edges]`. This can be in the order of millions.
+            edge_index (torch.Tensor): The edge indicies.
+            num_nodes (int, optional): The number of nodes in the graph.
+                (default: :obj:`None`)
+
+        Returns:
+            A node embedding tensor of shape :obj:`[num_nodes, out_channels]`.
+        """
+        num_nodes = maybe_num_nodes(edge_index, num_nodes)
+        time_info = self.temporal_encoder(edge_time)
+        edge_attr_time = torch.cat((time_info, edge_attr), dim=1)
+        edge_attr_time = self.temporal_encoder_head(edge_attr_time)
+
+        # `to_dense_batch` assumes sorted inputs
+        if not self.is_sorted:
+            edge_index[1], indices = index_sort(edge_index[1])
+            edge_attr_time = edge_attr_time[indices]
+
+        # zero-pad each node's edges:
+        # [num_edges, hidden_channels] -> [num_nodes*K, hidden_channels]
+        edge_attr_time, _ = to_dense_batch(
+            edge_attr_time,
+            edge_index[1],
+            max_num_nodes=self.K,
+            batch_size=num_nodes,
```

Comment:
It looks like we are choosing `self.K` links, but it doesn't look like they are the latest.

Reply:
I'm not sure what you mean by "latest". The parameter `K` is a fixed parameter that represents the max number of edges each node can have. It is not dependent on each mini-batch size.

Reply:
That's a good point. If a destination node has degree larger than `K`, we will need to include `K` more recent edges, i.e., largest `K` elements in `edge_time` for the destination node.